apps/extension/src/Approvals/ApproveSignTx/ApproveSignTx.tsx

```diff
@@ -69,10 +90,61 @@ export const ApproveSignTx: React.FC<Props> = ({ setDetails }) => {
 
   return (
     <Stack className="text-white" gap={4}>
+      <div>
+        {/* TODO: Use an icon: */}
+        <ActionButton
+          onClick={() => navigate(TopLevelRoute.ApproveSignTxDetails)}
+        >
+          View Data
+        </ActionButton>
+      </div>
       <Alert type="warning">
         Sign this {accountType === AccountType.Ledger ? "Ledger " : ""}
         transaction?
       </Alert>
+      {details?.txDetails?.commitments &&
```

Comment:
I think we could separate this switch from the body. But we're gonna work in the ux anyways, so no worries about that for now o/

Reply:
yeah, I didn't like this either, but didn't want to impose much structure since I knew you would look at it :)